### PR TITLE
clarify perfect_power behavior

### LINKED ISSUE #17048: Behavior of perfect_power when candidates is given but has no valid exponents
See #17044.

```
>>> from sympy import *
>>> print(perfect_power(216000, [2]))
(60, 3)
```

This should either be documented or the behavior should be changed to return False.

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

closes #17048 
#### Release Notes

- ntheory
    - clarify `perfect_power` behavior in docstring

### CODE REVIEW

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -319,6 +321,32 @@ def perfect_power(n, candidates=None, big=True, factor=True):
...
     [(0, False), (1, True), (2, True), (3, False), (4, True)]
+
+    It is not necessary to provide candidates. But if you do then
+    be aware that a false negative result will be returned if the
+    provided candidates are too large:
+
+    >>> perfect_power(60**3, [20])
+    False
+
+    To only test for exponents in candidates, use False for ``factor``:
```
Comment (smichr) [inline sympy/ntheory/factor_.py]:
```suggestion
    To only test for exponents in ``candidates``, use False for ``factor``:
```

File: sympy/ntheory/factor_.py [hunk 2]
```diff
@@ -319,6 +321,32 @@ def perfect_power(n, candidates=None, big=True, factor=True):
     >>> is2pow = lambda n: bool(n and not n & (n - 1))
     >>> [(i, is2pow(i)) for i in range(5)]
     [(0, False), (1, True), (2, True), (3, False), (4, True)]
+
+    It is not necessary to provide candidates. But if you do then
+    be aware that a false negative result will be returned if the
+    provided candidates are too large:
+
+    >>> perfect_power(60**3, [20])
```
Comment (smichr) [inline sympy/ntheory/factor_.py]:
```suggestion
    >>> perfect_power(60**3, [20])  # 2**20 > 60**3 so 20 is impossibly large
```

**Comment by smichr** [comment]
ping @sylee957

**Comment by oscargus** [comment]
Looks good! But will let @sylee957 have his say.

**Comment by sylee957** [comment]
I think that my original opinion was toward making the `candidates` to become strict candidates, than making them like a loose hint.
Would there be any difficulty making the code changes to preserve the behavior 9 years ago?

**Comment by smichr** [comment]
> Would there be any difficulty making the code changes to preserve the behavior 9 years ago?

What is it that bothers you about the current behavior: getting False instead of None?

File: sympy/ntheory/factor_.py [hunk 3]
```diff
@@ -372,11 +385,16 @@ def perfect_power(n, candidates=None, big=True, factor=True):
...
+                        for i in candidates:
+                            if not e % i:
+                                return r**(e//i), i
+                        return failure
                     e0 = primefactors(e)
                     if len(e0) > 1 or e0[0] != e:
                         e0 = e0[0]
                         r, e = r**(e//e0), e0
-                return r, e
+                    return r, e
```
Comment (smichr) [inline sympy/ntheory/factor_.py]:
```suggestion
                return r, e
```

**Comment by sylee957** [comment]
> > Would there be any difficulty making the code changes to preserve the behavior 9 years ago?
> 
> What is it that bothers you about the current behavior: getting False instead of None?

No, I mean that exponents can be strictly choosen from the candidates, and give `False` if there is no hint from the candidates.
I think that it would be confusing to make a single candidate under `log(n, 2) + 2` succeeding regardless of how the hint is tied closely to the actual answer.

**Comment by smichr** [comment]
@sylee957 , please see if this is more like what you were thinking.

**Pull request closed without merging**